trailblazer/store/api.py

```diff
@@ -475,6 +486,41 @@ def update_run_status(self, analysis_id: int, ssh: bool = False) -> None:
             analysis_obj.status = "error"
             self.commit()
 
+    @staticmethod
+    def query_tower(config_file: str, case_id: str):
+        # Currently only one tower ID is supported
+        tower_id = safe_load(open(config_file)).get(case_id)[0]
+        tower_api = TowerAPI(executor_id=tower_id)
+        if not tower_api.tower_client.meets_requirements:
+            raise TowerRequirementsError
+        return tower_api
+
+    def update_tower_run_status(self, analysis_id: int) -> None:
+        """Query tower for entries related to given analysis, and update the Trailblazer database"""
+        analysis_obj = self.analysis(analysis_id)
+        tower_api = self.query_tower(
+            config_file=analysis_obj.config_path, case_id=analysis_obj.family
+        )
+        try:
+            analysis_obj.status = tower_api.status
+            analysis_obj.progress = tower_api.progress
+            analysis_obj.logged_at = dt.datetime.now()
+            self.update_tower_jobs(
+                analysis_obj=analysis_obj, jobs=tower_api.get_jobs(analysis_id=analysis_obj.id)
+            )
+            self.commit()
+        except Exception as error:
+            LOG.error(f"Error logging case - {analysis_obj.family} : {error.__class__.__name__}")
```

Comment:
Usually you try to avoid accessing  attributes with "__". Any public methods that you can use to log instead?

Reply:
changing to this for now: type(error).__name__